### PR TITLE
Bug/fix resizing 598

### LINKED ISSUE #598: Fix Resizing of Component in flow Container at different break points
## Description
Issue 1:
Main issue is that w-full isn't added by default.
Leading to bugs such as https://github.com/johnsonandjohnson/Bodiless-JS/issues/327
Issue 2:
We don't have the all the breakpoints at tablet size for resizing.
Issue 3:
When resizing, width is removed when resizing ends
Issue 4:
Fix default width to use classes instead of %'s.

### Steps to reproduce

Clear steps describing how to reproduce the issue. Please please please link to a demo project if possible, this makes your issue _much_ easier to diagnose (seriously).
### Expected result

What should happen?

### Actual result

What happened.

### Environment

Run `gatsby info --clipboard` in your project directory and paste the output here.

### Affected Version

- If the issue was encountered on a site, please attach your `package.json`
- If the issue was encountered in the Bodiless-JS repo, please link to a
  commit or branch on which the issue occurs.

### PR DESCRIPTION
**Pull request by e2thex**
## Changes

* Depricated FlowContainer prop defaultWidth
* Created FlowContainer prop defaultWidthClasses
* Ensure that class are added to the dom as things are resized
* Ensure that the width is removed when resizing ends

## Test Instructions

Verify in Edit/Review/Static
1. Verify anything added by default flowcontainer automatically gets 'w-full' 
1. Verify resizing anything in the flowcontainer maintains w-full and size set (Desktop -- all sizes, Tablet - 50%, Mobile - none) 
1. Place a varity of container items 
* Row with 25% at desktop, 50% at tablet, 100% at mobile
* Row with 50% at tablet & desktop, 100 at mobile 
* Row with 100% at mobile, 50% at tablet, 100% at desktop

Verify Edit mode only:
1. Verify that constrained widths of 50/100 are still in effect. (Desktop, Tablet, Mobile)
1. Verify that limit is in effect
1. Verify that each of the default widths on placement holds true

Verify in Edit/Review/Static
1. Verify https://github.com/johnsonandjohnson/Bodiless-JS/issues/327
* Add 4 accordions and add titles
* leave first two 1005
* Resize to tablet and make 2 50% at tablet and thus desktop.
* All should render at 100% on mobile

Verify in Edit/Review/Static
* Add a new page.
* Add 3 image components to page
* Resize to tablet & resize each to 33%.
* Click out of edit & resize to desktop
* Click edit and they should be at 33%

## Related Issues

Fixes #598 

BREAKING CHANGE: 
This PR breaks the old FlowContainer API's prop defaultWidth, it uses to take a number and now needs to take a function that will return a string.

### CODE REVIEW

File: packages/bodiless-layouts/README.md [hunk 1]
```diff
@@ -164,14 +164,20 @@ There are two helper functions for `snapData`.
 
 ---
 
-#### Default Width
+#### Default Width (depricated)
```
Comment (hvanyo) [inline packages/bodiless-layouts/README.md:167]:
```suggestion
#### Default Width (deprecated)
```

File: packages/bodiless-layouts/src/SlateSortableResizable.tsx [hunk 2]
```diff
@@ -80,6 +80,7 @@ const SlateSortableResizable = (props: Props) => {
     useGetMenuOptions,
     ...rest
   } = props;
+  console.log(rest.className);
```
Comment (hvanyo) [inline packages/bodiless-layouts/src/SlateSortableResizable.tsx:83]: Can we remove before merging?

**Comment by hvanyo** [comment]
@e2thex 

There is one bug I see:

File: packages/bodiless-layouts/src/FlowContainer/utils/appendTailwindWidthClass.ts [hunk 3]
```diff
@@ -175,10 +175,13 @@ const defaultSnapData = getSnapFrom(
   withTuple('(min-width: 992px)')(75)('lg:w-3/4'),
   withTuple('(min-width: 992px)')(100)('lg:w-full'),
```
Comment (hvanyo) [inline packages/bodiless-layouts/src/FlowContainer/utils/appendTailwindWidthClass.ts:176]: Could we add sm supports both 25%, 33%, 66%, 75% by default here?  or every flow container we setup for sites porting need to define it.

const defaultSnapData = getSnapFrom(
  withTuple('(min-width: 0px)')(100)('w-full'),
  withTuple('(min-width: 576px)')(25)('sm:w-1/4'),
  withTuple('(min-width: 576px)')(33.33)('sm:w-1/3'),
  withTuple('(min-width: 576px)')(50)('sm:w-1/2'),
  withTuple('(min-width: 576px)')(66.66)('sm:w-2/3'),
  withTuple('(min-width: 576px)')(75)('sm:w-3/4'),
  withTuple('(min-width: 576px)')(100)('sm:w-full'),
  withTuple('(min-width: 992px)')(25)('lg:w-1/4'),
  withTuple('(min-width: 992px)')(33.33)('lg:w-1/3'),
  withTuple('(min-width: 992px)')(50)('lg:w-1/2'),
  withTuple('(min-width: 992px)')(66.66)('lg:w-2/3'),
  withTuple('(min-width: 992px)')(75)('lg:w-3/4'),
  withTuple('(min-width: 992px)')(100)('lg:w-full'),
);

Comment (e2thex) [reply packages/bodiless-layouts/src/FlowContainer/utils/appendTailwindWidthClass.ts:180]: I updated to have all of this options

**Comment by e2thex** [comment]
@hvanyo I fixed the bug you found

**Review by hvanyo — APPROVED** [review]

File: examples/test-site/src/data/pages/flow-container/index.tsx [hunk 4]
```diff
@@ -62,31 +62,32 @@ const FlowContainerPage = (props: any) => (
       <FlowContainerDefault
         id="width_25"
         nodeKey="width_25"
-        defaultWidth="25"
+        defaultWidth={asDefaultClasses('w-full lg:w-1/4')}
```
Comment (wodenx) [inline examples/test-site/src/data/pages/flow-container/index.tsx:65]: I don't think we need `asDefaultClasses` - it made me look in some other place to see what was going on, and it's really no harder to write
```
defaultWidth={() => 'w-full lg:w-1/4'}
```

Comment (wodenx) [reply examples/test-site/src/data/pages/flow-container/index.tsx:65]: but maybe rename this prop to something like `getDefaultWidth` ?

File: examples/test-site/src/data/pages/flow-container/index.tsx [hunk 5]
```diff
@@ -62,31 +62,32 @@ const FlowContainerPage = (props: any) => (
...
       />
       <h2 className="text-2xl font-bold mt-4">Default Width of 33% (should round up to 33.33%)</h2>
       <FlowContainerDefault
         id="width_33"
         nodeKey="width_33"
-        defaultWidth="33"
+        defaultWidthClasses="w-full lg:w-1/3"
```
Comment (wodenx) [inline examples/test-site/src/data/pages/flow-container/index.tsx:71]: Should this be removed?

File: packages/bodiless-layouts/README.md [hunk 6]
```diff
@@ -166,11 +166,16 @@ There are two helper functions for `snapData`.
 
 #### Default Width
 
-You can set the default width for a new item via the `defaultWidth` prop. This should be a number
-from 0-100 expressing the desired default width as a percentage. The actual default width will be
-the smallest *allowed* width which is greater than the value specified.  For example, if you set
-allowed widths of 50% and 100%, specifying `defaultWidth="51"` will give you an actual default width
-of 100%.
+One can set the default width classes via the `defaultWidth` prop.  The prop is a function that will 
+be pass the snapData function.  It is expect to return a string of the starting classes
```
Comment (wodenx) [inline packages/bodiless-layouts/README.md:170]: "it is expected"

File: packages/bodiless-layouts/README.md [hunk 7]
```diff
@@ -144,7 +144,7 @@ There are two helper functions for `snapData`.
 
    const snapData = withtailwindClasses(tailwindConfig)('w-full, md:w-full, md:w-1/2, lg:w-full, lg:w-1/2, lg:w-1/3, lg:w-1/4');
    ```
----
+   ---
```
Comment (wodenx) [inline packages/bodiless-layouts/README.md:147]: Is this intentional?

File: packages/bodiless-layouts/README.md (hunk 6 above)
Comment (wodenx) [inline packages/bodiless-layouts/README.md:170]:
```suggestion
be passed the snapData function.  It is expected to return a string of the starting classes
```

File: packages/bodiless-layouts/src/FlowContainer/SortableChild.tsx [hunk 8]
```diff
@@ -36,28 +41,22 @@ const SortableChild = (props: SortableChildProps) => {
...
   const [snapWidth, setSnapWidth] = useState('');
-  // Store what size is being used we set the width so that re-sizeable uses percent when resizeing
-  // we default the size to what the current class width is.
-  // we can not set this to auto becaus then it resizes in pixels and not percent.
+  // We start this off as not set so that the classes are used
   const [size, setSize] = useState({
     height: '',
-    width: `${snap({ className: snapClassName }).width}%`,
+    width: '',
+    // width: `${snap({ className: snapClassName }).width}%`,
```
Comment (wodenx) [inline packages/bodiless-layouts/src/FlowContainer/SortableChild.tsx:59]: Should this comment be removed?

File: examples/test-site/src/data/pages/flow-container/index.tsx (hunk 4 above)
Comment (e2thex) [reply examples/test-site/src/data/pages/flow-container/index.tsx:65]: Renamed and removed asDefaultClassess (also fixed in in the flow-container file

File: packages/bodiless-layouts/README.md (hunk 6 above)
Comment (e2thex) [reply packages/bodiless-layouts/README.md:170]: done

File: packages/bodiless-layouts/README.md (hunk 7 above)
Comment (e2thex) [reply packages/bodiless-layouts/README.md:147]: fixed

File: packages/bodiless-layouts/src/FlowContainer/SortableChild.tsx (hunk 8 above)
Comment (e2thex) [reply packages/bodiless-layouts/src/FlowContainer/SortableChild.tsx:59]: fixed

**Review by wodenx — APPROVED** [review]

**Review by hvanyo — APPROVED** [review]

**Pull request merged**